src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,263 @@
+use super::*;
+
+pub struct Go {}
+
+impl Go {
+  /// Generate golang code using tokens from parser
+  pub fn generate(lb: &mut LangBuilder, t: AnilizedTokens) -> Result<(), LocationError> {
+    // TODO: Replace when file importing is implemented
+    lb.code("package main");
+
+    let mut code = Self {};
+
+    // define functions
+    for (_, func) in t.functions {
+      code.function(func, lb);
+    }
+
+    // define types
+    for (_, type_) in t.types {
+      code.code_type(type_, lb);
+    }
+
+    // define structs
+    for (_, structure) in t.structs {
+      code.structure(structure, lb);
+    }
+
+    // define globals
+    for (_, glob) in t.vars {
+      code.global_var(glob, lb);
+    }
+
+    Ok(())
+  }
+  /// Get the type as a string
+  pub fn get_type(&mut self, type_: Option<Type>) -> String {
+    match type_ {
+      Some(res) => match res.type_ {
+        TypeType::Char => "rune".to_string(),
+        TypeType::String => "string".to_string(),
+        TypeType::Struct(res) => self.type_struct(res),
+        TypeType::Int => "int".to_string(),
+        TypeType::I8 => "int8".to_string(),
+        TypeType::I16 => "int16".to_string(),
+        TypeType::I32 => "int32".to_string(),
+        TypeType::I64 => "int64".to_string(),
+        TypeType::UInt => "uint".to_string(),
+        TypeType::U8 => "uint8".to_string(),
+        TypeType::U16 => "uint16".to_string(),
+        TypeType::U32 => "uint32".to_string(),
+        TypeType::U64 => "uint64".to_string(),
+        TypeType::TypeRef(res) => res,
+        TypeType::Enum(_) | TypeType::Array(_) => unimplemented!(),
```

Comment:
I think we want to support the array type in GoLang as it's very often used.

Reply:
Yeah, I'm not quite sure how this would be implemented as I'm new to enums and the golang implementation of enums seems quite unusual.